### PR TITLE
[API-33] Occurrence type and status endpoints

### PR DESCRIPTION
**Pull request by quetzyg**
## Description
This pull request addresses issue [API-33](https://github.com/vostpt/api/issues/33).

## Task items:
- [X] Added `GET` `/occurrences/families` endpoint;
- [X] Added `GET` `/occurrences/species` endpoint;
- [X] Added `GET` `/occurrences/types` endpoint;
- [X] Added `GET` `/occurrences/statuses` endpoint;
- [X] Added test coverage for new endpoints;
- [X] Document new endpoints;

### CODE REVIEW

File: routes/api/occurrences.php [hunk 1]
```diff
@@ -18,6 +22,27 @@
...
+        OccurrenceTypeController::class,
+        'index',
+    ])->name('types::index');
+
+    Route::get('/statuses', [
+        OccurrenceStatusController::class,
+        'index',
+    ])->name('statuses::index');
+
+
```
Comment (tiagojpdias) [inline routes/api/occurrences.php]: Intended extra newline?

File: tests/Integration/Controllers/OccurrenceSpeciesController/IndexEndpointTest.php [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                    ],
+                ],
+            ],
+        ]);
+    }
+
+    /**
+     * @test
+     */
+    public function itSuccessfullyIndexeSpeciesOccurrences(): void
```
Comment (tiagojpdias) [inline tests/Integration/Controllers/OccurrenceSpeciesController/IndexEndpointTest.php]: `Indexe` typo?

Comment (quetzyg) [reply tests/Integration/Controllers/OccurrenceSpeciesController/IndexEndpointTest.php]: Thanks, I saw this one when looking at the test output. Already sorted.

File: routes/api/occurrences.php (hunk 1 above)
Comment (quetzyg) [reply routes/api/occurrences.php]: Sorted

**Pull request merged**